packages/truffle-box/box.js

```diff
@@ -66,13 +70,35 @@ const Box = {
     }
   },
 
+  checkDir: async (options = {}, destination) => {
+    let logger = options.logger || console;
+    if (!options.force) {
+      const unboxDir = fs.readdirSync(destination);
+      if (unboxDir.length) {
+        logger.log(`This directory is non-empty...`);
+        const prompt = [
+          {
+            type: "confirm",
+            name: "proceed",
+            message: `Proceed anyway?`,
+            default: true
+          }
+        ];
+        const answer = await inquirer.prompt(prompt);
+        if (!answer.proceed) {
+          logger.log("Unbox cancelled");
+          process.exit();
```

Comment:
Is this how we should we handle the case where they do not want to proceed? We don't pass the callback down this far so maybe this is best.

Reply:
Yeah, thought about that when moving the logic. I don't see the harm in handling it like this, but I'm not opposed to it being resolved otherwise.

Reply:
Well, breaking the entire command flow this far away from `cli.js` just feels like a code smell to me. It probably won't hurt anything, like you said. I'll take another look and see if there is something that can be done that isn't too involved.